hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/BaseWriteHelper.java

```diff
@@ -61,6 +61,7 @@ public HoodieWriteMetadata<O> write(String instantTime,
         // perform index loop up to get existing location of records
         context.setJobStatus(this.getClass().getSimpleName(), "Tagging: " + table.getConfig().getTableName());
         taggedRecords = tag(dedupedRecords, context, table);
+        context.clearJobStatus();
```

Comment:
If lazy execution happens afterwards, the job status may not be properly populated.  Have you verified all places that this won't happen?